lib/tapioca/gem/pipeline.rb

```diff
@@ -156,7 +162,12 @@ def on_constant(event)
         return if T::Enum === constant # T::Enum instances are defined via `compile_enums`
 
         mark_seen(name)
-        compile_constant(name, constant)
+
+        if event.is_a?(Gem::NotOwnedConstantFound)
+          compile_not_owned_constant(name, T.cast(constant, Module))
```

Comment:
If you use `event.constant` you won't need to cast it:

```suggestion
          compile_not_owned_constant(name, event.constant)
```

I think we could refactor a bit the code to not use a local variable.